### PR TITLE
Enable and fix linting

### PR DESCRIPTION
**Pull request by rahulPalliyalil**
### What

This pr attempts to enable golangcilint in this repo and fix all the existing lint issues

### How to review

Please check if the changes make sense and all the expected linters are enabled. 

### Who can review

Anyone except me

### CODE REVIEW

File: interceptor/interceptor.go [hunk 1]
```diff
@@ -66,16 +58,6 @@ var pathsToUse = []string{
 	"/v1/instances",
 }
 
-// Check to see whether the response should be remapped
-func shallIgnore(path string) bool {
```
Comment (nshumoogum) [inline interceptor/interceptor.go:70]: I don't believe this should be removed

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: interceptor/interceptor.go (hunk 1 above)
Comment (rahulPalliyalil) [reply interceptor/interceptor.go:70]: this is not being used and the linter complained as well ?

Comment (nshumoogum) [reply interceptor/interceptor.go:70]: I will double check outside of this PR but found this https://github.com/ONSdigital/dp-api-router/pull/171/files

Comment (rahulPalliyalil) [reply interceptor/interceptor.go:70]: sorry didn't understand, the link you send is to my pr right ? btw this is an un exported function so can't be used outside the package and we don't have consumers for it in the package. As i mentioned the linter complained that to be unused as well.

Comment (nshumoogum) [reply interceptor/interceptor.go:70]: Sorry it was this line: https://github.com/ONSdigital/dp-api-router/pull/171/files#:~:text=if%20shallIgnore(r.URL.Path)%20%7B

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**